suite/tests/linux/syscall_pwait.cpp

```diff
@@ -278,31 +280,32 @@ main(int argc, char *argv[])
          * of the call and restore it post syscall. The mask check is making sure
          * that save/restore is done properly.
          */
-        asm volatile("mov %7, %%rbx\n\t"
-                     "movq %2, %%rax\n\t"
-                     "movq %3, %%rdi\n\t"
-                     "movq %4, %%rsi\n\t"
-                     "movq %5, %%rdx\n\t"
-                     "movq %6, %%r10\n\t"
-                     "movq %7, %%r8\n\t"
-                     "movq %8, %%r9\n\t"
-                     "syscall\n\t"
-                     "mov $0, %0\n\t"
-                     "cmp $-4095, %%rax\n\t"
-                     "jl no_syscall_error%=\n\t"
-                     "mov $1, %0\n\t"
-                     "no_syscall_error%=:\n\t"
-                     "mov $0, %1\n\t"
-                     "cmp %%rbx, %%r8\n\t"
-                     "je no_mask_error%=\n\t"
-                     "mov $1, %1\n\t"
-                     "no_mask_error%=:\n"
-                     /* early-clobber outputs */
-                     : "=&r"(syscall_error), "=&r"(mask_error)
-                     : "r"((int64_t)SYS_epoll_pwait), "rm"((int64_t)epoll_fd),
-                       "rm"(&events), "r"(24LL), "r"(-1LL), "rm"(&test_set),
-                       "r"((int64_t)(_NSIG / 8))
-                     : "rax", "rdi", "rsi", "rdx", "r10", "r8", "r9", "rbx");
+        asm volatile(
+            "mov %7, %%rbx\n\t"
+            "movq %[sys_num], %%rax\n\t"
+            "movl %[epfd], %%edi\n\t"
+            "movq %[events], %%rsi\n\t"
+            "movl %[maxevents], %%edx\n\t"
+            "movq %[timeout], %%r10\n\t"
+            "movq %[sigmask], %%r8\n\t"
+            "movq %[ss_len], %%r9\n\t"
+            "syscall\n\t"
+            "mov $0, %[syscall_error]\n\t"
+            "cmp $-4095, %%rax\n\t"
+            "jl no_syscall_error%=\n\t"
+            "mov $1, %[syscall_error]\n\t"
+            "no_syscall_error%=:\n\t"
+            "mov $0, %[mask_error]\n\t"
+            "cmp %%rbx, %%r8\n\t"
+            "je no_mask_error%=\n\t"
+            "mov $1, %[mask_error]\n\t"
+            "no_mask_error%=:\n"
+            /* early-clobber outputs */
+            : [syscall_error] "=&r"(syscall_error), [mask_error] "=&r"(mask_error)
+            : [sys_num] "rm"((long)SYS_epoll_pwait), [epfd] "rm"(epoll_fd),
```

Comment:
Repeating prior request: remove the long cast and write to %eax.

Reply:
I did see your prior request and I explicitly changed the 'int' into writing to 32-bit registers (even though the test works fine without it). This 'long' here was intentional because syscall() does take it as long, it is not needed here indeed. 'long' doesn't guarantee 64-bit AFAIK by the way so it wouldn't resolve a operand error of moving it to a 64-bit register in inline asm. The test should be safe now either way, I've removed it and write the 'int' to 32-bit reg now.